01_classify_stratified/nbconverted/plot_cancer_type_results.py

```diff
@@ -283,3 +285,136 @@ def label_points(x, y, gene, sig, ax):
 plt.title('Cancer type confusion matrix, methylation data', size=14, pad=14)
 plt.tight_layout()
 
+
+# How to read the confusion matrices:
+# 
+# * Remember, we're doing one-vs-rest classification of the cancer types. The cancer type on the y-axis is what the model is trained to predict (i.e. the positive label in the training set), and the cancer type on the x-axis is what we're evaluating on (i.e. the true label in the test set).
```

Comment:
Just to clarify, the first column in the bottom row is showing the accuracy of a model to distinguish ACC vs other cancer types, where the model was trained to distinguish UVM from other types of cancers? So a low performance value indicates that the methylation for UVM isn't able to predict ACC. So there isn't shared methylation information between these two cancer types?

Reply:
Close - it's showing the accuracy of a model trained to distinguish ACC vs. other cancer types on test samples whose true label is UVM. If the model was perfectly accurate (always classified ACC samples and other samples correctly), there would be 1 on the diagonal and 0 everywhere else, since the model would only predict UVM samples as 1 and everything else as 0.

So a low value indicates that it's rare for the model to "confuse" UVM with ACC, suggesting that their methylation patterns probably aren't very similar.

Does that make more sense? It's a bit of a counterintuitive way to define similarity between cancer types, since we're using supervised classifiers to do it (normally you'd just use some sort of unsupervised distance metric). But we had the data already, so we decided it's a good place to start.

Reply:
So a low value means that your UVM-trained model could not successfully distinguish between ACC vs other cancers samples? Suggesting that the methylation patterns are not similar?